src/include/common/object_pool.h

```diff
@@ -76,17 +104,72 @@ class ObjectPool {
   /**
    * Returns a piece of memory to hold an object of T.
    *
+   * It throws exception if the object pool fails to fetch memory.
```

Comment:
@throw or @throws ? I don't remember, but there's a directive you should use for these things.